Lib/unittest/async_case.py

```diff
@@ -90,9 +90,12 @@ def _callSetUp(self):
         self._callAsync(self.asyncSetUp)
 
     def _callTestMethod(self, method):
-        if self._callMaybeAsync(method) is not None:
-            warnings.warn(f'It is deprecated to return a value that is not None from a '
-                          f'test case ({method})', DeprecationWarning, stacklevel=4)
+        if (result := self._callMaybeAsync(method)) is not None:
```

Comment:
Please do not use `:=`  without need. Additional parentheses and longer line makes it harder to read.
```suggestion
        result = self._callMaybeAsync(method)
        if result is not None:
```